version: '3.9'

services:
  backendwithfrontend:
    container_name: backendwithfrontend
    build:
      context: .
      dockerfile: Dockerfile-local
    ports:
      - 6200:8080
    environment:
      - COGNITO_DOMAIN=${COGNITO_DOMAIN}
      - CLIENT_ID=${CLIENT_ID}
      - CLIENT_SECRET=${CLIENT_SECRET}
      - COGNITO_USERPOOL_ID=${COGNITO_USERPOOL_ID}
      - REDIRECT_PATH=${REDIRECT_PATH}
    networks:
      - cognito-try-net
    command: java -jar app.jar

  dynamodb-local:
    container_name: dynamodb-local
    image: amazon/dynamodb-local
    ports:
      - '8085:8000'
    command: -jar DynamoDBLocal.jar -dbPath /data -sharedDb
    volumes:
      - ./dynamodb_data_cognito:/data # ボリュームマウントができず、バイマウントにした。。。
    networks:
      - cognito-try-net
    restart: unless-stopped

  dynamodb-admin:
    container_name: dynamodb-admin
    image: aaronshaf/dynamodb-admin:latest
    environment:
      - DYNAMO_ENDPOINT=http://dynamodb-local:8000
    ports:
      - 8001:8001
    depends_on:
      - dynamodb-local
    networks:
      - cognito-try-net
networks:
  cognito-try-net:
    driver: bridge


